### PR TITLE
Add a TruffleHog dectector for Kong Konnect Platform 

### PR DESCRIPTION
**Pull request by mrwanny**
### Description:

Add a TruffleHog dectector for Kong Konnect Platform (api.konghq.com) 

token format `(?:kpat_|spat_)[a-zA-Z0-9]{49}`

issue: https://github.com/trufflesecurity/trufflehog/issues/5306

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))?

---

> [!NOTE]
> **Low Risk**
> Additive detector and wiring only; verification uses outbound HTTP to Kong’s API with no changes to core scan or auth paths. Review whether `DetectorType_Kong` should remain in `excludedFromDefaultList` now that the scanner is in `buildDetectorList`.
> 
> **Overview**
> Adds **Kong Konnect** (`kpat_` / `spat_` tokens) as a new TruffleHog detector with optional live verification against `GET https://global.api.konghq.com/v3/organizations/me` (Bearer auth). **200** and **403** count as verified; **401** as invalid; other statuses surface verification errors. Verified hits can include a Konnect API rotation guide in `ExtraData`.
> 
> The scanner is registered in default detectors behind **`KongDetectorEnabled`** (on by default in `main.go`), with proto enum **`DetectorType_Kong` (1072)** and unit/integration tests for regex matching and verification edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b1a4215bfb027e9a9b5ed292fd742cc453ac4600. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: pkg/engine/defaults/defaults.go [hunk 1]
```diff
@@ -1904,6 +1906,8 @@ func buildDetectorList() []detectors.Detector {
 			return !feature.ResendDetectorEnabled.Load()
 		case *weightsandbiasesv2.Scanner:
 			return !feature.WeightsAndBiasesV2DetectorEnabled.Load()
+		case *kong.Scanner:
+			return !feature.KongDetectorEnabled.Load()
```
[the remark this answers is not in the document]
Comment (mrwanny) [reply pkg/engine/defaults/defaults.go:1910]: `main.go` has been updated

File: pkg/detectors/kong/kong.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	switch res.StatusCode {
+	case http.StatusOK:
+		// If the endpoint returns useful information, we can return it as a map.
+		return true, extraData, nil
+	case http.StatusUnauthorized:
+		// The secret is determinately not verified (nothing to do)
+		return false, nil, nil
+	default:
+		return false, nil, fmt.Errorf("unexpected HTTP response status %d", res.StatusCode)
+	}
```
[the remark this answers is not in the document]
Comment (mrwanny) [reply pkg/detectors/kong/kong.go:103]: 403 are now handled
